### PR TITLE
feat: add change password

### LINKED ISSUE #740: Add option to change password to unlock
**Is your feature request related to a problem? Please describe.**
I want to change my Alby password, but there is no option to do so.

**Describe the solution you'd like**
Change password under settings. After login. Prompt for current password, then enter new password.

**Are you working on this? (Yes/No)**
No

### PR DESCRIPTION
**Pull request by im-adithya**
### Describe the changes you have made in this PR

Adds option to change password

### Link this PR to an issue

Closes #740

### Type of change (Remove other not matching type)

- `feat`: New feature (non-breaking change which adds functionality)

### Screenshots of the changes (If any)

Screenshot from 2022-06-13 23-22-51
Screenshot from 2022-06-13 23-22-55

### How has this been tested?

Have to add tests, which would be done after making necessary changes if requested!

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

**Comment by escapedcat** [comment]
Didn't check in detail yet but it looks like this exactly copies over [the functionality from the onboarding screen](https://github.com/pseudozach/lightning-browser-extension/blob/blocklist/src/app/screens/Onboard/SetPassword/index.tsx).  
Is there a way to generalize this? I.e. cut components out and reuse these? Would be great if we can avoid maintaining i.e. password rules/validations in two places.

**Comment by bumi** [comment]
was thinking the same. Can we somehow unify that? or unify the password validation?

**Comment by bumi** [comment]
what is the status on this? do we still want to extract the validation? @im-adithya @escapedcat

**Comment by im-adithya** [comment]
Ready for review!

File: src/app/components/PasswordForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import {
+  HiddenIcon,
+  VisibleIcon,
+} from "@bitcoin-design/bitcoin-icons-react/outline";
+import TextField from "@components/form/TextField";
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+
+export type Props = {
+  keyPrefix: string;
```
Comment (escapedcat) [inline src/app/components/PasswordForm/index.tsx:10]: Should we name this `i18nKeyPrefix`? So people only looking at the props know what it's for?

File: src/app/screens/Settings.tsx [hunk 2]
```diff
@@ -228,6 +251,67 @@ function Settings() {
...
+          closeTimeoutMS={200}
+          isOpen={modalIsOpen}
+          onRequestClose={closeModal}
+          contentLabel="Edit account name"
+          overlayClassName="bg-black bg-opacity-25 fixed inset-0 flex justify-center items-center p-5"
+          className="rounded-lg bg-white w-full max-w-lg"
+        >
+          <div className="p-5 flex justify-between dark:bg-surface-02dp">
+            <h2 className="text-2xl font-bold dark:text-white">
+              Change Password
```
Comment (escapedcat) [inline src/app/screens/Settings.tsx:281]: Can we use i18n strings for the whole PR? Or in a follow-up PR?

File: src/app/components/PasswordForm/index.tsx (hunk 1 above)
Comment (im-adithya) [reply src/app/components/PasswordForm/index.tsx]: Done!

**Comment by escapedcat** [comment]
Tested and works. Not sure if we should show some sort of confirm-message after the password was changed. I.e. one of our toasts. But could be done later on as well.

File: src/app/components/PasswordForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+  i18nKeyPrefix,
+}: Props) {
+  const [errors, setErrors] = useState(initialErrors);
+  const [passwordView, setPasswordView] = useState(false);
+  const [passwordConfirmationView, setPasswordConfirmationView] =
+    useState(false);
+  const { t } = useTranslation("translation", { keyPrefix: i18nKeyPrefix });
+
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    if (event.target.name === "password") {
```
Comment (bumi) [inline src/app/components/PasswordForm/index.tsx:39]: do we need these checks?

what happens if we do something like: 
```
formData[event.target.name] = event.target.value.trim();
setFormData(formData);
```

File: src/app/components/PasswordForm/index.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      event.target.name === "passwordConfirmation" &&
+      errors.passwordConfirmation &&
+      formData.password === event.target.value.trim()
+    ) {
+      setErrors({ ...errors, passwordConfirmation: "" });
+    }
+  }
+
+  function validate() {
+    let password = "";
```
Comment (bumi) [inline src/app/components/PasswordForm/index.tsx:64]: can we call this passwordErrorMessage/passwordConfirmationErrorMessage to be a bit more descriptive and indicate that we actually set the error messages here (and not the actual password field values.

File: src/app/screens/Settings.tsx [hunk 5]
```diff
@@ -34,6 +46,17 @@ function Settings() {
...
 
+  function closeModal() {
+    setModalIsOpen(false);
+  }
+
+  async function updateAccountPassword(password: string) {
+    await utils.call("changePassword", {
+      password: formData.password,
+    });
+    closeModal();
```
Comment (bumi) [inline src/app/screens/Settings.tsx:57]: I think adding a toast message her with a confirmation that the password was changed would be good. 
To make sure the user is not wondering if it worked or not.

**Review by escapedcat — APPROVED** [review]

**Review by bumi — CHANGES_REQUESTED** [review]
I think we should move the change password option to the general settings in the top.

In the "personal data" section we talk about "Payees can request for additional data to be sent with a payment." all this data is sent in the lnurl request.

**Comment by im-adithya** [comment]
Done!
Screenshot from 2022-07-21 16-01-18
Screenshot from 2022-07-21 16-01-36

File: src/app/screens/Settings.tsx [hunk 6]
```diff
@@ -182,6 +206,26 @@ function Settings() {
             </div>
           )}
         </Setting>
+
+        <Setting
+          title="Change Password"
```
Comment (escapedcat) [inline src/app/screens/Settings.tsx:211]: This is missing the i18 string. The label button below as well... sorry

Comment (im-adithya) [reply src/app/screens/Settings.tsx:211]: Oops, sorry! Done!

**Review by bumi — APPROVED** [review]

**Pull request merged**